Shop_Unity_base/Assets/Scripts/Shop/Model/Items/PotionFactory.cs

```diff
@@ -0,0 +1,62 @@
+﻿
+namespace Model.Items
+{
+    using System;
+
+    public class PotionFactory : IItemFactory
+    {
+        public enum PotionType { Health, Mana, Strength, Defence, Invisibility }
+
+        private Potion defaultHealth;
+        private Potion defaultMana;
+        private Potion defaultStrength;
+        private Potion defaultDefence;
+        private Potion defaultInvisibility;
```

Comment:
Overall, I feel as if the class could benefit from using a dictionary where the Key is the PotionType enum and the Value is the default potion of the relevant enum type. This way you can make it modifiable at the very least without having to add too much additional code, i.e. in CreatePotion(...) and the Create<type>Potion() methods. (you're committing some WET coding here).

Before you @ me, yes, I know that Switches are more performant than Dictionary look-ups. So what? Readability is usually the better option than performance unless you really do need the micro-optimization.

If you want to have it even more extendable and modular, you'd go into Reflection and even more abstract classes. Though that's probably overkill.

Reply:
We don't care about performance here :')
Dictionary lookup, I like it!